### ISSUE TITLE
err is an array, and not single error

### ISSUE THREAD

**Issue by iamstarkov** [desmondmorris/node-twitter#100]
I have such code:

```
  client.get('/users/show.json', options, function (err, info, raw) {
    console.log(err);
    console.log(err.message);
```

And here is a log:
```
[ { code: 135, message: 'Timestamp out of bounds.' } ]
undefined
```

why err is an array?

**Comment by iamstarkov** [comment]
@desmondmorris @jdub can you help?

**Comment by desmondmorris** [comment]
@iamstarkov What is in your options object?

**Comment by iamstarkov** [comment]
this:
```js
{ screen_name: 'jsunderhood',
  trim_user: true,
  count: 200,
  include_rts: true,
  exclude_replies: false };
```

**Comment by desmondmorris** [comment]
@iamstarkov is your system clock up to date?

https://twittercommunity.com/t/timestamp-out-of-bounds/287/2

**Comment by iamstarkov** [comment]
no-no, its not the point of the issue

**Comment by iamstarkov** [comment]
the point is that `err` param passing to my callback is not an error object as expected, but array of them

**Comment by desmondmorris** [comment]
@iamstarkov gotcha - apologies

**Comment by desmondmorris** [comment]
@iamstarkov I think this is culprit - https://github.com/desmondmorris/node-twitter/blob/master/lib/twitter.js#L164.

**Comment by iamstarkov** [comment]
can you tell me why it is an array. cannot figure out it from source code

**Comment by desmondmorris** [comment]
@iamstarkov the "errors" property is appended to the response directly from the API.  I assume, there are cases where the API would return more than one error at a time.

**Comment by iamstarkov** [comment]
it contradicts with common node-way to pass only one error in callback and to [docs](https://github.com/desmondmorris/node-twitter#rest-api).

I thought about multiply errors and found that rest API can return only one error in response.

**Comment by iamstarkov** [comment]
@desmondmorris what do you think?

**Comment by desmondmorris** [comment]
@iamstarkov I agree with you.  A potential solution would be pulling out the custom error handling, passing only the error object from the request.  The client would then be responsible for parsing errors out of the response object.

https://github.com/desmondmorris/node-twitter/blob/master/lib/twitter.js#L147-L177  would be come:


```js
this.request(options, function(error, response, data){
  if (error) {
    return callback(error, data, response);
  }
  
  try {
    data = JSON.parse(data);
  }
  catch(parseError) {
    error = parseError
  }

  callback(error, data, response);
});
````

thoughts?

**Comment by iamstarkov** [comment]
looks good

**Comment by iamstarkov** [comment]
@desmondmorris i can create pull-request, if you want

**Comment by iamstarkov** [comment]
I found this document https://dev.twitter.com/ads/basics/response-codes and there is only one error per request. Going to create an patch

**Comment by qawemlilo** [comment]
Thanks guys for fixing this, it also took me a bit of time to debug.

**Comment by iamstarkov** [comment]
@qawemlilo patch is not landed yet #107. considering to use [twit](https://www.npmjs.com/package/twit)
